### PR TITLE
Make `build-wasm.sh` work for macOS

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)

I confirm that this PR (choose one, and delete the other options):

- [x] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.

### CODE REVIEW

**Review by cdisselkoen — APPROVED** [review]

**Comment by cdisselkoen** [comment]
Approved if it passes CI, but the current version fails CI with a `sed` error

File: cedar-wasm/build-wasm.sh [hunk 1]
```diff
@@ -62,21 +67,21 @@ fix_package_json_files() {
...
-    sed -i "s/[{]\s*>=: /{ \">=\": /g" "$types_file"
-    sed -i "s/[{]\s*&&: /{ \"\&\&\": /g" "$types_file"
-    sed -i "s/[{]\s*||: /{ \"||\": /g" "$types_file"
-    sed -i "s/[{]\s*[+]: /{ \"+\": /g" "$types_file"
-    sed -i "s/[{]\s*-: /{ \"-\": /g" "$types_file"
-    sed -i "s/[{]\s*[*]: /{ \"*\": /g" "$types_file"
-    sed -i "s/[{]\s*\.: /{ \".\": /g" "$types_file"
-    sed -i "s/ | __skip//g" "$types_file"
-    sed -i "s/ { .\+: __skip } |//g" "$types_file"
+    sed -i '' "s/[{]\s*!: /{ \"!\": /g" "$types_file"
```
Comment (john-h-kastner-aws) [inline cedar-wasm/build-wasm.sh:70]: For mac/linux portability, IIRC, I've had to write `-i'.bak'` (no space after `-i`). Then you'll want to include a `rm `"$types_file".bak` to clean up.

Comment (john-h-kastner-aws) [reply cedar-wasm/build-wasm.sh:70]: also, we  could do this with one `sed` command:

```bash
sed -i'.bak' "
s/[{]\s*!: /{ \"!\": /g;
s/[{]\s*==: /{ \"==\": /g;
...
" "$types_file"
```

Comment (shaobo-he-aws) [reply cedar-wasm/build-wasm.sh]: Thanks for the tip. Done.

**Review by john-h-kastner-aws — APPROVED** [review]

**Pull request merged**